### ISSUE TITLE
Don’t show direct messages on own profile

### ISSUE THREAD

**Issue by nclm** [mastodon/mastodon#23414]
### Steps to reproduce the problem

1. Write and send a direct message to someone
2. Visit your own profile

### Expected behaviour

The direct message is shown in your profile’s timeline, under “Posts”

### Actual behaviour

The direct message shouldn’t be there?

### Detailed description

I understand that direct messages are just posts with a specific privacy setting. I also understand that they are not really “private”. But as a user, they feel different and should not be treated just like any posts.

Seeing direct messages on your profile makes you think you posted them publicly by mistake.

It also provides no way to get a clean view of your profile since it’s full of direct messages.

Two possibilities:
- Posts that are set as “Mentioned people only” should not show on profiles
- Posts that starts with a mention should not show on profiles

That’s for the “Posts” tab. The “Posts and replies” is a different matter. Here, public posts starting with a mention should definitely appear. For posts that are  “Mentioned people only”, I can see both ways working.

### Specifications

Mastodon v4.0.2

**Comment by trwnh** [comment]
if i go to a profile i expect to see all posts that i have access to. i am really uncomfortable with the idea that my own posts would become invisible to me, and far more uncomfortable with the idea that simply starting a post with a mention is enough to do this (necessitating the old ".@" behaviour of twitter).

with that said, i think the confusion about "posts" vs "posts and replies" is more to do with how arbitrary the division is. i think it would make more sense to have a single "posts" tab with different filters you can apply, like in #10011. although i am not sure if there's any justification for treating direct posts any differently. they don't feel any different to me... a post is a post. i frequently reply to higher-scoped posts by dropping to direct visibility. it is also possible to increase the scope later.

basically if the expectation is that "direct messages" should be separate, then we should actually have "messages" that are a separate thing from posts.

**Comment by eishiya** [comment]
I'd go one step further in clarifying the fact that "mentioned users only" posts are not DMs are remove the "Direct messages" section from the web client entirely. It serves no specific purpose since the posts are visible in your notifications (and possibly home timeline) and messages the user wants to easily access later can be bookmarked. The DM view only seems to confuse users into thinking "DMs" are something different from regular posts, and cause them to treat this privacy setting as something special. Users are effectively giving up a useful privacy setting when they treat it as Twitter-style DMs, something you don't send without invitation, something that disrespects the recipient's privacy somehow :/

**Comment by nclm** [comment]
I actually agree with both of you about how we should not treat these messages differently.

The idea of refactoring the profile page so that it’s not the “Posts”/“Posts and replies” tabs but something more granular could work well (as long as it’s still simple enough), and the idea of removing the DM view since it’s duplicating Notifications and confuses what these messages really is makes sense too. Possibly the notifications could be also split into two views, “Notifications” (without mentions), and “Mentions” (replacing DMs).

But for the profile pages, I still find it confusing that the default view on my own profile page includes “Mentioned people only” that starts with a mention. I realise that I separated these before, but it’s actually the combination.

Somehow, when I click on my profile, I expect to see what most people will see on my profile. And including messages I sent directly to other people isn’t really part of my “profile”, it’s simply stuff that I happened to have used my account to post. Of course they should be easy retrievable by filtering in “Direct mentions” or similar on the profile, just like you could do with Replies.

I’ll try to describe a more precise behaviour some day. But I think it’s simply about the default filtering on profiles.

**Comment by eishiya** [comment]
I usually use my profile as a way to quickly find specific posts of my own, so I usually want to see these extra posts. However, an option to see what strangers would see would be useful. So, in addition to the aforementioned post type settings (posts, replies, boosts, media...), it would be interesting to have settings for access levels when viewing your own profile - self, followers, non-followers.

**Comment by trwnh** [comment]
> remove the "Direct messages" section entirely

i do agree it was probably a mistake to ever call them "direct messages" and pretend they were something they were not. see also:

- #18186
- #3819
- #19950

> Possibly the notifications could be also split into two views, “Notifications” (without mentions), and “Mentions” (replacing DMs)?

we kind of have something better already with the quick filter bar

> Somehow, when I click on my profile, I expect to see what most people will see on my profile.

> an option to see what strangers would see would be useful

there is something like that on facebook and youtube, i think? you can view your own profile as if you were someone else. technically you could open a private tab/window to view the logged-out representation?
